### ISSUE TITLE
Unnecessary dropout condition in LSTM model

### ISSUE THREAD

**Issue by ghost** [pytorch/fairseq#895]
https://github.com/pytorch/fairseq/blob/69d0f7f826e58a7e07bc431afd74ce2c54c63eb6/fairseq/models/lstm.py#L200

"if num_layers > 1 else 0." is unnecessary since PyTorch 1.0 onward will not introduce dropout for the last layer.

Reference: 
https://pytorch.org/docs/1.0.0/nn.html?highlight=lstm#torch.nn.LSTM
https://pytorch.org/docs/stable/nn.html?highlight=lstm#torch.nn.LSTM

**Comment by lematt1991** [comment]
Happy to take PR!

**Comment by lematt1991** [comment]
Closing due to inactivity
